wled00/FX_fcn.cpp

```diff
@@ -213,24 +213,12 @@ CRGBPalette16 &Segment::loadPalette(CRGBPalette16 &targetPalette, uint8_t pal) {
   if (pal < 245 && pal > GRADIENT_PALETTE_COUNT+13) pal = 0;
   if (pal > 245 && (strip.customPalettes.size() == 0 || 255U-pal > strip.customPalettes.size()-1)) pal = 0; // TODO remove strip dependency by moving customPalettes out of strip
   //default palette. Differs depending on effect
-  if (pal == 0) switch (mode) {
```

Comment:
The unfortunate consequence of this removal is that when used from existing preset, there will be no "default" palette to choose from. So, unless user chooses the effect in UI (with defined "default" palette from metadata) it will not have "default" palette defined. Once user selects it from UI the palette will be defined.

This is then inconsistent behaviour.

Reply:
the default palette gets set when FX is loaded.
I don't fully understand in what scenario the behaviour would differ (with the latest bugfix). Can you check latest commit changes and give me a description of a test case I can try?

Reply:
Take, for example, effect "Fire 2012". Its default platte is "Fire" (35) even if user selects palette "Default" (0).

If I have an existing preset with Fire 2012 it will most likely have `”pal”:0` as its selected palette. But after removal of this code this will mean "Party" palette until user manually selects Fire 2012 from UI which will populate its default palette to 35.

Steps to reproduce:
- use 0.14 to create preset of Fire 2012 with default values (i.e. palette chosen is Default)
- update to this PR
- restart ESP
- select preset with Fire 2012 effect
- it will display with Party colors instead of Fire
- select another preset or effect
- now select Fire 2012 in Effects tab, palette will change to Fire
- select another preset or effect
- select preset with Fire 2012
- it will display with Fire palette.

Reply:
Got it, I will try to fix this.

Reply:
ok, fixed it. can confirm your test scenario now works.